Application/src/model/Parser.java

```diff
@@ -99,66 +130,65 @@ public void parseSequenceDiagram(String inputJSON) {
                 classNamesString = (tempClassNames.substring(0, tempClassNames.length() - 1));
             }
 
-            // get the elements of the First Diagram
-            for (Messages diagramElement : parsedSequenceDiagram.getDiagram().getContent().get(0).getMessages()) {
-                tempFirstDiagram += "{" + diagramElement.getFrom() + "," + diagramElement.getTo() + ",[" + diagramElement.getMessage().get(0) + ", " +
-                        diagramElement.getMessage().get(1) + ", " + diagramElement.getMessage().get(2) + "]" + "}" + ",";
-                //removing extra comma
-                firstDiagramString = (tempFirstDiagram.substring(0, tempFirstDiagram.length() - 1));
-            }
-
+            diagram = parseMessages(parsedSequenceDiagram.getDiagram().getContent().get(0).getMessages());
+            diagram = wrap(diagram);
 
-            // get the elements of the Parallel Diagram
-            for (Messages parallelDiagramElement : parsedSequenceDiagram.getDiagram().getContent().get(1).getMessages()) {
-                tempParallelDiagram += "{" + parallelDiagramElement.getFrom() + "," + parallelDiagramElement.getTo() + ",[" + parallelDiagramElement.getMessage().get(0) + "]" + "}" + ",";
-                //removing extra comma
-                parallelDiagramString = (tempParallelDiagram.substring(0, tempParallelDiagram.length() - 1));
+            // If there is a parallel diagram.
+            if (parsedSequenceDiagram.getDiagram().getContent().size() > 1) {
+                parallel = parseMessages(parsedSequenceDiagram.getDiagram().getContent().get(1).getMessages());
```

Comment:
i think this is fine, but @timjon  expressed his concern about the possibility of there being more than one parallel diagram. we should bring this up in the next physical meeting